93 add heater step parameters

eugenebertin opened the pull request:
1. added heater step parameters
2. tweaked the deposition_parameter subsection to only include namely deposition related parameters
3. more process information has been added to their own subsections (SourceRampUp, SourcePresput, SourceDepRate, SulfurCrackerPressure)
4. many small improvement to the base Lf_event class, overview plot and optix spectrum data handling
5. implemented the handling of N2, NH3 and O2 reactive gases 
6. implemented flags for sputtering process for which something went wrong
7. implemented the plot of the position of the sample in the chamber

hampusnasstrom commented:
Looks good, I'll have to do the detailed review a bit later. One thing to keep in mind is that changing the names of quantities means we will need to reprocess old entries but we should be able to do this from the CLI on the server. We should also double check if these are used in the app.

eugenebertin commented:
I will continue developing a little on this branch a little. So I guess we can wait a little for the review @hampusnasstrom

eugenebertin commented:
1. added heater step parameters
2. tweaked the deposition_parameter subsection to only include namely deposition related parameters
3. more process information has been added to their own subsections (SourceRampUp, SourcePresput, SourceDepRate, SulfurCrackerPressure)
4. many small improvement to the base Lf_event class, overview plot and optix spectrum data handling
5. implemented the handling of N2, NH3 and O2 reactive gases

eugenebertin commented:
Hi @hampusnasstrom, have you had a little bit of time to process my pull request ? thank you

hampusnasstrom reviewed (commented):
I think I would need some more information to properly review this and I won't have time to look through the 1352(!) new lines of code in sputter log reader. Glancing through it looks okay. The only thing I'm a little bit concerned about is the changing of names. We need to make sure that these are not used anywhere or we might have something failing on the production server.

eugenebertin commented:
@hampusnasstrom, I think we are all set set to complete the pull request

hampusnasstrom reviewed (approved):
Looks good! The archive is now getting quite big and takes some time to load so that might be the next thing to tackle.
